src/app/wallets/update-pending-invoices.ts

```diff
@@ -78,158 +82,170 @@ const updatePendingInvoice = async ({
-    return false
-  }
-
-  const receivedBtc = paymentAmountFromNumber({
-    amount: roundedDownReceived,
-    currency: WalletCurrency.Btc,
-  })
-  if (receivedBtc instanceof Error) return receivedBtc
-
-  const walletInvoiceReceiver = await WalletInvoiceReceiver({
-    walletInvoice,
-    receivedBtc,
-    usdFromBtc: dealer.getCentsFromSatsForImmediateBuy,
-    usdFromBtcMidPrice: usdFromBtcMidPriceFn,
-  })
-  if (walletInvoiceReceiver instanceof Error) return walletInvoiceReceiver
-
-  const lockService = LockService()
-  return lockService.lockPaymentHash(paymentHash, async () => {
-    // we're getting the invoice another time, now behind the lock, to avoid potential race condition
-    const invoiceToUpdate = await walletInvoicesRepo.findByPaymentHash(paymentHash)
-    if (invoiceToUpdate instanceof CouldNotFindError) {
-      pendingInvoiceLogger.error({ paymentHash }, "WalletInvoice doesn't exist")
-      return false
-    }
-    if (invoiceToUpdate instanceof Error) return invoiceToUpdate
-    if (walletInvoice.paid) {
-      pendingInvoiceLogger.info("invoice has already been processed")
-      return true
-    }
-
-    const displayCurrencyPerSat = await getCurrentPrice()
-    if (displayCurrencyPerSat instanceof Error) return displayCurrencyPerSat
-
-    if (!lnInvoiceLookup.isSettled) {
-      const invoiceSettled = await lndService.settleInvoice({ pubkey, secret })
-      if (invoiceSettled instanceof Error) return invoiceSettled
-    }
-
-    const invoicePaid = await walletInvoicesRepo.markAsPaid(paymentHash)
-    if (invoicePaid instanceof Error) return invoicePaid
-
-    // TODO: this should be a in a mongodb transaction session with the ledger transaction below
-    // markAsPaid could be done after the transaction, but we should in that case not only look
-    // for walletInvoicesRepo, but also in the ledger to make sure in case the process crash in this
-    // loop that an eventual consistency doesn't lead to a double credit
-
-    const metadata = LedgerFacade.LnReceiveLedgerMetadata({
-      paymentHash,
-      fee: walletInvoiceReceiver.btcBankFee,
-      feeDisplayCurrency: Number(
-        walletInvoiceReceiver.usdBankFee.amount,
-      ) as DisplayCurrencyBaseAmount,
-      amountDisplayCurrency: Number(
-        walletInvoiceReceiver.usdToCreditReceiver.amount,
-      ) as DisplayCurrencyBaseAmount,
-      pubkey: walletInvoiceReceiver.pubkey,
-    })
-
-    //TODO: add displayCurrency: displayPaymentAmount.currency,
-    const result = await LedgerFacade.recordReceive({
-      description,
-      recipientWalletDescriptor: walletInvoiceReceiver.recipientWalletDescriptor,
-      amountToCreditReceiver: {
-        usd: walletInvoiceReceiver.usdToCreditReceiver,
-        btc: walletInvoiceReceiver.btcToCreditReceiver,
-      },
-      bankFee: {
-        usd: walletInvoiceReceiver.usdBankFee,
-        btc: walletInvoiceReceiver.btcBankFee,
+}): Promise<boolean | ApplicationError> =>
+  asyncRunInSpan(
```

Comment:
please use `wrapAsyncToRunInSpan` and then `addAttributesToCurrentSpan`

Reply:
Ah are we doing this now instead of using `asyncRunInSpan`?

Reply:
Done: https://github.com/GaloyMoney/galoy/pull/1526/commits/50c2399e9e6f0f7d095eb88a1f2d289bbba7ec05